From 86d4eeb0592849acf70d0a06885350f2858d0060 Mon Sep 17 00:00:00 2001
From: Ben Bright <ben.bright@twdb.texas.gov>
Date: Fri, 17 Feb 2023 18:47:50 +0000
Subject: [PATCH] Changes for patch 0004-Welcome-Email-on-Invite-Acceptance on
 2.9.8

---
 ckan/views/user.py | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/ckan/views/user.py b/ckan/views/user.py
index b09b2faff4..2684a79c0d 100644
--- a/ckan/views/user.py
+++ b/ckan/views/user.py
@@ -746,6 +746,23 @@ class PerformResetView(MethodView):
                     {u"id": user_dict[u'id'], u"state": model.State.ACTIVE}
                 )
             mailer.create_reset_key(context[u'user_obj'])
+
+            # if user_state is 'pending', this was a response to an invite,
+            # so a welcome message is sent
+            if user_state == 'pending':
+                name = user_dict['fullname'] or user_dict['name']
+                email = user_dict['email']
+                subject = 'Welcome to the Texas Water Data Hub'
+                extra_vars = {
+                    'login_link': '/user/login',
+                    'site_title': config.get('ckan.site_title'),
+                    'site_url': config.get('ckan.site_url'),
+                    'user_name': name
+                }
+                body = base.render('emails/welcome_user.txt', extra_vars)
+                body_html = base.render('emails/welcome_user.html', extra_vars)
+                mailer.mail_recipient(name, email, subject, body, body_html)
+
             h.flash_success(_(u'Your password has been reset.'))
             return h.redirect_to(u'home.index')
         except logic.NotAuthorized:
-- 
2.25.1

